### PR TITLE
Strip \\?\ prefix from path generated by resolveSymlink

### PR DESCRIPTION
**Pull request by ahoppen**
This aligns the implementation of resolveSymlinks with that of Foundation.String._resolvingSymlinksInPath, which received this canonicalization fix in https://github.com/swiftlang/swift-foundation/pull/639.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/TSCBasic/PathShims.swift [hunk 1]
```diff
@@ -24,20 +24,21 @@ import Foundation
...
+        GetFinalPathNameByHandleW(handle, nil, 0, DWORD(VOLUME_NAME_DOS))
     return try withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength)) {
         guard GetFinalPathNameByHandleW(handle, $0.baseAddress!, DWORD($0.count),
-                                        DWORD(FILE_NAME_NORMALIZED)) == dwLength - 1 else {
+                                        DWORD(VOLUME_NAME_DOS)) == dwLength - 1 else {
             throw FileSystemError(.unknownOSError, path)
         }
-        let path = String(decodingCString: $0.baseAddress!, as: UTF16.self)
+        // When using `VOLUME_NAME_DOS`, the returned path uses `\\?\`.
+        let path = String(decodingCString: $0.baseAddress!.advanced(by: 4), as: UTF16.self)
```
Comment (compnerd) [inline Sources/TSCBasic/PathShims.swift:41]: Do we want an assertion that the prefix is present?

**Review by compnerd — APPROVED** [review]

File: Sources/TSCBasic/PathShims.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/TSCBasic/PathShims.swift]: I was following what you did in https://github.com/swiftlang/swift-foundation/pull/639. I guess we should also add the check there.

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/TSCBasic/PathShims.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/TSCBasic/PathShims.swift]: Making the corresponding change to swift-foundation here: https://github.com/swiftlang/swift-foundation/pull/974

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
macOS CI failure is unrelated. This change only affects Windows. macOS CI is failing because this is the only repo that build SourceKit-LSP using a 5.9 host compiler, but SourceKit-LSP requires 5.10 to build successfully. We didn’t hit this before because this repo hasn’t seen any changes in the last 2 months, which is when SourceKit-LSP started requiring Swift 5.10.

**Pull request merged**